lib/envutil.rb

```diff
@@ -228,6 +228,21 @@ def flush; end
   end
   module_function :verbose_warning
 
+  if Warning.respond_to?(:[]=)
```

Comment:
Sorry, the `required_ruby_version` is still 2.3 that is before `Warning` module.

```suggestion
  if defined?(Warning.[]=)
```